cabal.project.release

```diff
@@ -1,11 +1,23 @@
-packages: ./ghcup.cabal
-
-optional-packages: ./vendored/*/*.cabal
+import: cabal.project
 
 optimization: 2
 
-package ghcup
-  flags: +tui -tar
+if impl(ghc < 9.0)
```

Comment:
This is not correct. The release project file always wants to build with `-tar`. The tar flag is only there to circumvent complicated errors during development, which sometimes happens due to libarchive.

Reply:
Ah, OK. I'll drop a note in a commit reverting this, let me know if you want me to squash them together (so the resulting commit will just add a comment noting this discrepancy vs the other cabal.project files)

Reply:
Also, is this why the `mingw32` flags are present in `cabal.project` (and will be imported in `cabal.project.release`, whereas the `linux`, `darwin`, and `freebsd` flags are only present in `cabal.project.release`? Windows causes more trouble that needs to be caught at dev time, but the others are mostly just optimizations for release time?